// ng6-social-button/esm5/lib/buttons/linkedinSharingButton.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
import { Component, Input, Output, EventEmitter, Inject, ElementRef, ViewChild } from '@angular/core';
import { DOCUMENT } from '@angular/platform-browser';
import { SocialService } from '../social.service';
var LinkedinSharingButton = /** @class */ (function () {
    function LinkedinSharingButton(socialAuthService, document) {
        this.socialAuthService = socialAuthService;
        this.document = document;
        this.share = { href: location.href };
        this.response = new EventEmitter();
    }
    /**
     * @return {?}
     */
    LinkedinSharingButton.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        // add linkedin share button script tag to element
        this.element.nativeElement.innerHTML = "<script type=\"IN/Share\" data-url=\"" + this.share.href + "\"></script>";
        // render share button
        // window['IN'] && window['IN'].parse();
    };
    LinkedinSharingButton.decorators = [
        { type: Component, args: [{
                    selector: 'linkedin-share-button',
                    template: "\n        <span #element></span>\n    ",
                },] },
    ];
    /** @nocollapse */
    LinkedinSharingButton.ctorParameters = function () { return [
        { type: SocialService },
        { type: undefined, decorators: [{ type: Inject, args: [DOCUMENT,] }] }
    ]; };
    LinkedinSharingButton.propDecorators = {
        share: [{ type: Input }],
        response: [{ type: Output }],
        element: [{ type: ViewChild, args: ['element',] }]
    };
    return LinkedinSharingButton;
}());
export { LinkedinSharingButton };
if (false) {
    /** @type {?} */
    LinkedinSharingButton.prototype.share;
    /** @type {?} */
    LinkedinSharingButton.prototype.response;
    /** @type {?} */
    LinkedinSharingButton.prototype.element;
    /** @type {?} */
    LinkedinSharingButton.prototype.socialAuthService;
    /** @type {?} */
    LinkedinSharingButton.prototype.document;
}
